python/utils/LinkedLibraryHolder.cpp

```diff
@@ -30,6 +30,30 @@ constexpr static const char NVQIR_SIMULATION_BACKEND[] =
     "NVQIR_SIMULATION_BACKEND=";
 constexpr static const char TARGET_DESCRIPTION[] = "TARGET_DESCRIPTION=";
 
+int countGPUs() {
+  char buffer[1024];
+  std::string output;
+  FILE *fp1, *fp2;
+
+  fp1 = popen("nvidia-smi", "r");
```

Comment:
```suggestion
  std::ifstream fp1("nvidia-smi");
```

Reply:
This change didn't yield the expected result, may be I'm missing something?